src/test.rs

```diff
@@ -34,12 +40,16 @@ const JWKS: &'static str = r#"{
 const AUDIENCE: &'static str =
     "37772117408-qjqo9hca513pdcunumt7gk08ii6te8is.apps.googleusercontent.com";
 
-struct TestKeyProvider;
+#[derive(Default)]
+struct TestKeyProvider {
+    call_count: Arc<std::sync::RwLock<u8>>,
```

Comment:
This isn't used anywhere. Perhaps you intended to check it in test_deadlock_prevention.